CondTools/SiPhase2Tracker/plugins/DTCCablingMapTestReader.cc

```diff
@@ -60,17 +62,15 @@ void DTCCablingMapTestReader::fillDescriptions(edm::ConfigurationDescriptions& d
   descriptions.add("DTCCablingMapTestReader", desc);
 }
 
-DTCCablingMapTestReader::DTCCablingMapTestReader(const edm::ParameterSet& iConfig) {}
+DTCCablingMapTestReader::DTCCablingMapTestReader(const edm::ParameterSet& iConfig) : cablingMapToken_(esConsumes()) {}
 
 DTCCablingMapTestReader::~DTCCablingMapTestReader() {}
 
 void DTCCablingMapTestReader::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetup) {
   using namespace edm;
   using namespace std;
 
-  edm::ESHandle<TrackerDetToDTCELinkCablingMap> cablingMapHandle;
-  iSetup.get<TrackerDetToDTCELinkCablingMapRcd>().get(cablingMapHandle);
-  TrackerDetToDTCELinkCablingMap const* p_cablingMap = cablingMapHandle.product();
+  const auto p_cablingMap = &iSetup.getData(cablingMapToken_);
```

Comment:
it was not checked before.

Reply:
I would say, adding the check to avoid unexpected behaviour would be ideal. However, if there is a reason not to check, it would be good to note (and to learn for me). Thanks.

Reply:
I am not sure what you want me to do here?
This is recommended pattern, see [here](https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#Getting_data_from_EventSetup_wit).
If the caller wants to check whether the data product exists, I could do:
```C++
 if(auto handle = iEventSetup.getHandle(cablingMapToken_)) {
    auto const& p_cablingmap = *handle;
  }
  else {
    edm::LogError("DTCCablingMapTestReader") << "missing cabling map"
  }
```
Note that in case the Record of the data product is missing, an exception is thrown, but this is seems to be an overkill. This is a standalone reader module to check offline conditions and if there is an error it means the Global Tag is misconfigured.

Reply:
I think that throwing an exception for a missing data product, as it is currently written, is the correct behavior here.